libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -506,23 +505,21 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
 
         ifNonNull(accessToken) { token ->
-            runInTelemetryContext { telemetry ->
-                logD(
-                    "MapboxMetricsReporter.init from MapboxNavigation main",
-                    telemetry.LOG_CATEGORY
-                )
-                MapboxMetricsReporter.init(
-                    navigationOptions.applicationContext,
-                    token,
-                    obtainUserAgent()
-                )
-                MapboxMetricsReporter.toggleLogging(navigationOptions.isDebugLoggingEnabled)
-                telemetry.initialize(
-                    this,
-                    navigationOptions,
-                    MapboxMetricsReporter,
-                )
-            }
+            logD(
+                "MapboxMetricsReporter.init from MapboxNavigation main",
+                MapboxNavigationTelemetry.LOG_CATEGORY
+            )
+            MapboxMetricsReporter.init(
```

Comment:
I didn't see the corresponding addition of flag check in `MapboxMetricsReporter`. Is this intentional?
Same for `toggleLogging` and `disable`.

Reply:
sorry, didn't get this one

Reply:
I think that's okay in this context. We shouldn't if-check on `init` because someone could flip the global setting back at any point in time. And the `TelemetryUtils.setEventsCollectionState` also disables the telemetry service under the hood so we don't need to check for that.